### ISSUE TITLE
#if bug

### ISSUE THREAD

**Issue by lestofante** [arduino/Arduino#1468]
```c
#ifdef __AVR_ATtinyX5__
const byte LED_PIN = 3;   // gamba 2
const byte RECV_PIN = 1;  // gamba 6
#else
const byte LED_PIN = 3;
const byte RECV_PIN = 10;
#endif
```
give error. declaring a variable before #if resolve the problem.

bug found by nid69ita

**Comment by ffissore** [comment]
Please give a try to the IDEs linked at the bottom of this email on the dev list
https://groups.google.com/a/arduino.cc/forum/#!msg/developers/4X2T3rCxXWM/YNJl6PZuDucJ
We're testing a new preprocessor and it compiles your sketch just fine.

**Comment by ffissore** [comment]
New preprocessor tracked at #2636. Builds for testing it are available

**Comment by ffissore** [comment]
Fixed by #3779
